routes.py

```diff
@@ -0,0 +1,663 @@
+
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    # sprawdzam czy current_user jest zalogowany, TRUE przekierowywuje na stronę użytkownika
+    if current_user.is_authenticated:
+        return redirect(url_for('user', username=current_user.username))
+    user = current_user
+    form = LoginForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(username=form.username.data).first()
+        if user and user.check_password(form.password.data):
+            login_user(user, remember=form.remember_me.data)
+            next_page = request.args.get('next')
+            if not next_page or url_parse(next_page).netloc != '':
+                next_page = url_for('user', username=user.username)
+            flash('You are logged in', 'success')
+            return redirect(next_page)
+        else:
+            flash('Invalid username or password', 'error')
+    return render_template('login.html', form=form, user=user)
+
+
+# ścieżka strony użytkownika, która pokazuje dane konta, profile oraz pozwala tworzyć profile
+@app.route('/user/<username>', methods=['GET', 'POST'])
+@login_required
+def user(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    profiles = Profile.query.filter_by(user_id=user.id).all()
+    if len(profiles) == 0:
+        profiles = []
+    form = ProfileForm()
+    return render_template('user.html', profiles=profiles, form=form, user=user, get_server_short_name=get_server_short_name)
+
+
+# ścieżka pozwalająca tworzyć profile i przekierowywuje spowrotem na stronę użytkownika
+@app.route('/user/<username>/create_profile', methods=['GET', 'POST'])
+@login_required
+def create_profile(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    form = ProfileForm()
+    if form.validate_on_submit():
+        existing_profile = Profile.query.filter_by(
+            nickname=form.nickname.data, server=form.server.data).first()
+        if existing_profile is not None:
+            flash('A profile with this nickname already exists on this server.', 'error')
+        else:
+            new_profile = Profile(
+                nickname=form.nickname.data,
+                server=form.server.data,
+                division=form.division.data,
+                best_position=form.pref_role.data,
+                alternative_position=form.alternative_role.data,
+                user_id=user.id
+            )
+            db.session.add(new_profile)
+            db.session.commit()
+            flash('Your profile has been created successfully.', 'success')
+    return redirect(url_for('user', username=user.username))
+
+
+# ścieżka edycji profilu
+@app.route('/user/<username>/<int:profile_id>/edit_profile', methods=['GET', 'POST'])
+@login_required
+def edit_profile(username, profile_id):
+    user = User.query.filter_by(username=username).first_or_404()
+    profile = Profile.query.filter_by(id=profile_id).first_or_404()
```

Comment:
Are you sure that this profile belong to the user that is currently logged in?

Reply:
yes, another user can access the edit page of another user's profile but will have a blank page